lib/plugins/blocks.js

```diff
@@ -227,6 +227,77 @@ function inject (bot, { version, storageBuilder }) {
     return bot.blockAt(blocks[0])
   }
 
+  bot.findBlocks2 = async (opts) => {
+    assert(
+      typeof opts.wantedBlock === 'number' || typeof opts.wantedBlocks === 'object',
+      Error('wantedBlock has to be an id of a block or wantedBlocks has to be an array of block ids')
+    )
+    assert(
+      typeof opts.wantedBlock !== 'number' || typeof opts.wantedBlocks !== 'object',
+      Error('Either set wantedBlock or wantedBlocks, not both')
+    )
+    assert(
+      typeof opts.matcher === 'function' || opts.matcher === undefined,
+      Error('Matcher must be a function')
+    )
+    assert(
+      opts.maxMatches > 0 || opts.maxMatches === undefined,
+      Error('Max blocks must be >0')
+    )
+    assert(
+      opts.searchRadius > 0 || opts.searchRadius === undefined,
+      Error('Search Radius must be >0')
+    )
+    assert(
+      opts.point instanceof Vec3 || opts.point === undefined,
+      Error('Start point must be a vec3 point')
+    )
+    const internalWantedBlocks = opts.wantedBlocks || [opts.wantedBlock]
+    const searchRadius = opts.searchRadius || 16
+
+    const foundBlocks = []
+    const { x: originX, y: originY, z: originZ } = (opts.point || bot.entity.position).floored()
```

Comment:
`floored()` creates an unnecessary temporary Vec3 instance. Creating temporary objects is bad for performances.